### PR TITLE
docs: Explicitly mention 'cargo install' in README

### PR DESCRIPTION
**Pull request by BatmanAoD**
For users who already have Rust/Cargo installed, installation with Cargo may be the preferred approach.

#### Description
The "Install Latest Version" section now contains two subsections, "From binary" (for the existing curlbash command) and "From source (with Rust)" (for the new command).

#### Motivation and Context
Since attempting to go "off the beaten path" when installing software often leads to unintended problems, I installed Starship using a recommended approach (`brew`) before attempting to install with Cargo, even though I generally prefer Cargo. Assuming there is no special reason _not_ to recommend Cargo, I think it's worth mentioning it in the README so that other users don't avoid it unnecessarily.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation of existing feature

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

I just `brew unlink`ed my first install, then ran `cargo install starship` and confirmed that Starship worked (after re-sourcing my Bash config to update the `PROMPT_COMMAND` and `PS1` to use the new binary path).

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,12 +138,18 @@ The prompt shows information you need while you're working, while staying sleek
 
    #### Install Latest Version
 
-   ##### With Shell:
```
Comment (matchai) [inline README.md]: I think this should remain "With Shell", so we can add a "With PowerShell" heading when #690 is completed.

**Comment by BatmanAoD** [comment]
@chipbuster Looks like it was deleted when `install.sh` was added: https://github.com/starship/starship/pull/643

File: README.md (hunk 1 above)
Comment (BatmanAoD) [reply README.md]: How about `Prebuilt Binary, With Shell`?

Comment (matchai) [reply README.md]: `Prebuilt binary with Shell` works for me. 👍

**Comment by BatmanAoD** [comment]
@matchai Better?

I couldn't see whether there's currently a minimum supported Rust version; if there is, that may be worth mentioning, but it could also be a hassle to keep it up-to-date.

**Pull request merged**